### DISCUSSION TITLE
Rate limit webhook option

### DISCUSSION THREAD

**Discussion by joshdchang** [directus/directus#13199] [Feature Requests]
It would be useful, as to not trigger too many unnecessary builds on hosting platforms such as Cloudflare Pages, to have the option in Directus to limit a specific webhook to a given rate (1 per 4 hours, 1 per day, etc...).

**Comment by benhaynes** [comment]
@rijkvanzanten is there a way to build a "Throttle" operation for Data Flows? Maybe by checking the last time a flow ran?

**Reply by rijkvanzanten** [reply]
No. It's also going to be incredibly difficult for us to agnostically decide which call of the hook is "unnecessary" enough to skip in a throttle 🤔 

I think the better approach here is to offer enough configuration so the user can decide themselves when the flow fires the webhook (through filters) and/or rely on the receiving endpoint to decide when to build or not.

**Reply by joshdchang** [reply]
The idea is not necessarily to agnostically decide whether a webhook call is unnecessary but rather to limit calls to the webhook based on a fixed time interval (i.e, if the webhook was called by Directus in the last hour, cancel that specific webhook request and that’s it). Not sure whether that is feasible or if that clarifies what I was requesting, and thank you guys for responding so quickly.

**Reply by rijkvanzanten** [reply]
Yeah that makes sense! But that still means we have to decide which of the webhooks to fire to the receiving end. The main concern there is that the payload can be different between webhook invocations, so it's a little difficult to just ignore a handful of them 🤔

**Comment by joshdchang** [comment]
The idea is not necessarily to agnostically decide whether a webhook call
is unnecessary but rather to limit calls to the webhook based on a fixed
time interval (i.e, if the webhook has called in the last hour, cancel that
specific webhook request and that’s it). Not sure if that clarifies what I
was requesting, and thank you guys for responding so quickly, I really love
Directus.

**Reply by azrikahar** [reply]
This seems to be almost identical to this comment https://github.com/directus/directus/discussions/13199#discussioncomment-2723484, hence I've marked it as duplicate. Do let me know if it's not intended as such (could be it was meant as a quote with a follow up message) 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
